src/contracts/middleware/BLSPubkeyRegistry.sol

```diff
@@ -0,0 +1,211 @@
+    }
+
+
+    function registerOperator(address operator, uint8[] memory quorumNumbers, BN254.G1Point memory pubkey) external onlyRegistryCoordinator returns(bytes32){
+        bytes32 pubkeyHash = BN254.hashG1Point(pubkey);
+        require(pubkeyHash != ZERO_PK_HASH, "BLSRegistry._registerOperator: cannot register zero pubkey");
+        require(quorumNumbers.length > 0, "BLSRegistry._registerOperator: must register for at least one quorum");
+        require(pubkeyCompendium.pubkeyHashToOperator(pubkeyHash) == operator,"BLSRegistry._registerOperator: operator does not own pubkey");
+        _processQuorumApkUpdate(quorumNumbers, pubkey, true);
+        bytes32 globalApkHash = _processGlobalApkUpdate(BN254.plus(_globalApk, pubkey));
+
+        emit RegistrationEvent(operator, pubkeyHash, globalApkHash);
+    }
+
+    /**
+     * @notice deregisters `operator` with the given `pubkey` for the quorums specified by `quorumBitmap`
+     * @dev Permissioned by RegistryCoordinator
+     */    
+    function deregisterOperator(address operator, uint8[] memory quorumNumbers, BN254.G1Point memory pubkey) external onlyRegistryCoordinator returns(bytes32){
+        bytes32 pubkeyHash = BN254.hashG1Point(pubkey);
+
+        require(quorumNumbers.length > 0, "BLSRegistry._deregisterOperator: must register for at least one quorum");
+        require(pubkeyCompendium.pubkeyHashToOperator(pubkeyHash) == operator,"BLSRegistry._deregisterOperator: operator does not own pubkey");
+
+
+        _processQuorumApkUpdate(quorumNumbers, pubkey, false);
+
+        bytes32 globalApkHash = _processGlobalApkUpdate(BN254.plus(_globalApk, BN254.negate(pubkey)));
+
+        emit DeregistrationEvent(operator, pubkeyHash, globalApkHash);
+    }
+
+    /// @notice returns the `ApkUpdate` struct at `index` in the list of APK updates for the `quorumNumber`
+    function getApkUpdateForQuorumByIndex(uint8 quorumNumber, uint256 index) external view returns (ApkUpdate memory){
+        return quorumToApkUpdates[quorumNumber][index];
+    }
+
+    /// @notice returns the `ApkUpdate` struct at `index` in the list of APK updates that keep track of the sum of the APK amonng all quorums
+    function globalApkUpdates(uint256 index) external view returns (ApkUpdate memory){
+        return globalApkUpdateList[index];
+    }
+
+    function quorumApk(uint8 quorumNumber) external view returns (BN254.G1Point memory){
+        return quorumToApk[quorumNumber];
+    }
+
+    function globalApk() external view returns (BN254.G1Point memory){
+        return _globalApk;
+    }
+
+
+    /**
+     * @notice get hash of the apk of `quorumNumber` at `blockNumber` using the provided `index`;
+     * called by checkSignatures in BLSSignatureChecker.sol.
+     */
+    function getApkHashForQuorumAtBlockNumberFromIndex(uint8 quorumNumber, uint32 blockNumber, uint256 index) external view returns (bytes32){
+        ApkUpdate memory quorumApkUpdate = quorumToApkUpdates[quorumNumber][index];
```

Comment:
Could you use the pattern shown [here](https://github.com/Layr-Labs/eigenlayer-contracts/blob/7f96a3a057e53004bf5c85dca447432c846934dc/src/contracts/middleware/RegistryBase.sol#L211) and [here](https://github.com/Layr-Labs/eigenlayer-contracts/blob/7f96a3a057e53004bf5c85dca447432c846934dc/src/contracts/middleware/RegistryBase.sol#L683). Way cleaner IMO.

Reply:
yup